### PR TITLE
 Support browser back button in Make offer flow

### PR DESCRIPTION
**Pull request by despo**
## Context
Currently when a user goes to the previous Make offer step using the browser back button, it causes the flow to break, as it only has knowledge of what the current step is through the `new` and `edit` offer controller actions.

I tried adding some system specs but I was getting some unrelated errors when enabling javascript. Any suggestions on how to approach this? Do we need to test this?

## Changes proposed in this pull request

Ensure that current step is also set as part of submit actions (`create` and `update`). This should enable the wizard to always return the correct next step and avoid any errors.

## Link to Trello card
https://trello.com/c/MjEArmnE/3617-investigate-supporting-browser-back-button-in-make-offer-flow

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/provider_interface/offer/conditions_controller.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ def new
       end
 
       def create
-        @wizard = OfferWizard.new(offer_store, conditions_attributes_for_wizard)
+        @wizard = OfferWizard.new(offer_store, conditions_attributes_for_wizard.merge!(current_step: 'conditions'))
```
Comment (benswannack) [inline app/controllers/provider_interface/offer/conditions_controller.rb:10]: Is it the case that the current step will be the same for all actions in each of the controllers? Could we move the `merge` to somewhere common?

**Comment by benswannack** [comment]
Can L27 of WizardPathHistory now be removed? Maybe we can enforce in there that a step is passed in?

**Review by cwrw — APPROVED** [review]
I had a look at the back button and implementations in specs, as Im wary of enabling JS for this as well. I found a method from Capybara but I couldn't get it to work locally: `go_back` as I think you need to register a driver for that to work, it doesn't with the current one, maybe something to look into briefly? But otherwise happy to keep as is also.

I tested locally and works perfectly! Really happy with the simple solution ⭐  
Past the suggestion around drying up merging the current step if possible (maybe in the params of the controller eg. `conditions_attributes_for_wizard`) looks good to me 👍 

I had a question around `OfferWizard.new` in the `offers_controller`, just wanted to double check we don't need to set anything there as well?

**Comment by despo** [comment]
@cwrw we shouldn't need to set anything up in the OffersController as in that instance the validation scope we use will always be 'save' (to ensure that all validations take place before submission).

**Review by benswannack — APPROVED** [review]
Nice!

**Pull request merged**